tools/c7n_mailer/c7n_mailer/azure/azure_queue_processor.py

```diff
@@ -110,6 +112,14 @@ def process_azure_queue_message(self, encoded_azure_queue_message):
         try:
             sendgrid_delivery = SendGridDelivery(self.config, self.logger)
             sendgrid_messages = sendgrid_delivery.get_to_addrs_sendgrid_messages_map(queue_message)
-            return sendgrid_delivery.sendgrid_handler(queue_message, sendgrid_messages)
+
+            if 'smtp_server' in self.config:
+                smtp_delivery = SmtpDelivery(config=self.config,
+                                             session=self.session,
+                                             logger=self.logger)
+                for to_addrs, message in six.iteritems(sendgrid_messages):
```

Comment:
nit actual send grid messages have a particular api client object model, if these are just email messages let's just call them that instead of sendgrid messages.